src/sampling/fibonacci.jl

```diff
@@ -0,0 +1,65 @@
+"""
+	FibonacciSampling(n)
+
+Generate samples on a sphere using the Fibonacci lattice method.
+The parameter `n` specifies the number of points to generate.
+In the regular Fibonacci lattice method, the number ϕ is the golden ratio ((1 + √5)/2),
+but different numbers can be used, preferably irrational.
+
+## Example
+
+Sample a sphere with 1000 points:
+
+```julia
+sample(Sphere((0,0,0), 1), FibonacciSampling(100))
+
+# sample using π instead of the golden ration
+sample(Box((0,0),(1,1)), FibonacciSampling(100,π))
+"""
+struct FibonacciSampling <: SamplingMethod
+  n::Int
+  ϕ::Real
+  function FibonacciSampling(n::Int, ϕ::Real=(1 + √5) / 2)
+    if n ≤ 0
+      throw(ArgumentError("Size must be positive"))
+    end
+    new(n, ϕ)
+  end
+end
+
+function sample(::AbstractRNG, geom::Geometry, method::FibonacciSampling)
+  # sample points in square
+  i = 0:(method.n - 1)
+  x, y = mod.(i / method.ϕ, 1), i / method.n
+
+  # get points for each geometry
+  ps = Iterators.map(zip(x, y)) do (x, y)
+    Point(_fibonaccipointstogeom(geom, x, y))
+  end
+
+  # get transformations to turn canonical geometry into current geometry
+  transforms = _fibonaccigeomtransforms(geom)
+
+  Iterators.map(ps) do p
+    p |> transforms
+  end
```

Comment:
I believe that this can be simplified to a simple loop where you construct the x[i], y[i], convert to u, v coordinates and then call geom(u, v). Try to avoid intermediate allocations with vectors of x and y.